lib/app/endpoint/home.rb

```diff
@@ -164,46 +171,45 @@ class Home < Endpoint
           end
         end
 
-        # Handles EHR Launch and any redirects
         get '/:id/:key/:endpoint/?' do
           instance = Inferno::Models::TestingInstance.get(params[:id])
-          halt 404 unless !instance.nil? && instance.client_endpoint_key == params[:key] && %w[launch redirect].include?(params[:endpoint])
+          halt 404 unless !instance.nil? && instance.client_endpoint_key == params[:key] && ['launch','redirect'].include?(params[:endpoint])
 
           sequence_result = instance.waiting_on_sequence
 
           if sequence_result.nil? || sequence_result.result != 'wait'
-            redirect "/#{base_path}/#{params[:id]}/?error=no_#{params[:endpoint]}"
+            redirect "/#{BASE_PATH}/#{params[:id]}/?error=no_#{params[:endpoint]}"
           else
-            klass = Inferno::Sequence::SequenceBase.subclasses.find do |x|
-              x.name.demodulize.start_with?(sequence_result.name)
-            end
+            klass = instance.module.sequences.find{|x| x.name.demodulize == sequence_result.name}
 
             client = FHIR::Client.new(instance.url)
-            client.use_dstu2
+            if instance.version == 'dstu2'
+              client.use_dstu2
+            end
             client.default_json
             sequence = klass.new(instance, client, settings.disable_tls_tests, sequence_result)
 
-            timer_count = 0
-            stayalive_timer_seconds = 20
+            timer_count = 0;
+            stayalive_timer_seconds = 20;
 
             stream do |out|
+
               EventMachine::PeriodicTimer.new(stayalive_timer_seconds) do
-                timer_count += 1
+                timer_count = timer_count + 1;
```

Comment:
The [default RuboCop configuration](https://rubocop.readthedocs.io/en/latest/cops_style/#styleselfassignment) prefers the shorthand for self-assignment:
```suggestion
                timer_count += 1
```
consider reverting or changing the RuboCop configuration for consistency by disabling `Style/SelfAssignment`